src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
+pub trait FileRepository {
+    fn read_template_file(&self, path: &TemplateLocation) -> Result<String>;
```

Comment:
Side note: languages like Rust are awesome but they always make me want more.  In cases like yours, transferring ownership is the optimal solution. In caching cases, borrows are ideal.  Its hard to satisfy both.  I guess that is what `Cow` is for.  Unsure if we should use `Cow` or not.